src/odemis/gui/cont/project.py

```diff
@@ -336,100 +349,174 @@ def _on_text(self, evt):
+                                                                                           self.calib_model.name.value)
 
 
 class FastEMCalibrationRegionsController(object):
     """
     Listens to the calibration buttons and creates the FastEMROCControllers accordingly.
     """
 
-    def __init__(self, tab_data, calibration_bar, view_ctrl):
+    def __init__(self, tab_data, calibration_panel, view_ctrl):
         """
-        tab_data (FastEMAcquisitionGUIData): tab data model
-        calibration_bar (FastEMCalibrationBar): main calibration panel
-        view_ctrl (FastEMAcquisitionViewport): viewport controller
+        :param tab_data (FastEMAcquisitionGUIData): The tab data model.
+        :param calibration_panel (FastEMCalibrationPanelHeader): The main calibration panel including the 9 regions
+                of calibration (ROC) buttons, the calibrate button, the gauge and the label.
+        :param view_ctrl (FastEMAcquisitionViewport): The viewport controller. TODO replace with just view
         """
-        self._view_ctrl = view_ctrl
-        self._calibration_bar = calibration_bar
         self._tab_data = tab_data
+        self._data_model = self._tab_data.main
+        self._calibration_panel = calibration_panel
+        self._view_ctrl = view_ctrl
 
-        self.panel = FastEMCalibrationPanel(calibration_bar, tab_data.main.scintillator_layout)
-        calibration_bar.add_calibration_panel(self.panel)
-
-        self.roc_ctrls = {}  # int --> FastEMROCController
+        self.panel = FastEMCalibrationPanel(calibration_panel, tab_data.main.scintillator_layout)
+        calibration_panel.add_calibration_panel(self.panel)
 
         for btn in self.panel.buttons.values():
-            btn.Bind(wx.EVT_BUTTON, self._on_button)
+            btn.Bind(wx.EVT_TOGGLEBUTTON, self._on_button)
             btn.Enable(False)  # disabled by default, need to select scintillator in chamber tab first
 
+        # create calibration controller for each scintillator
+        self.roc_ctrls = {}
+        for roc_num, roc in self._tab_data.calibration_regions.value.items():
+            self.roc_ctrls[roc_num] = FastEMROCController(roc_num, self._tab_data, self._view_ctrl)
+            roc.coordinates.subscribe(self._on_coordinates)
+
         # Only enable buttons for scintillators which have been selected in the chamber tab
         tab_data.main.active_scintillators.subscribe(self._on_active_scintillators)
 
+        tab_data.main.is_acquiring.subscribe(self._on_is_acquiring)  # enable/disable button
+
     def _on_button(self, evt):
-        btn = evt.GetEventObject()
-        btn.SetLabel("OK")
-        btn.SetForegroundColour(wx.GREEN)
+        """
+        Called when one of the 9 single region of calibration (ROC) buttons is triggered.
+        A ROC can be added or removed by clicking again. If a ROC is added, an overlay will be
```

Comment:
Maybe a typo: I pronounce ROC as "arr ooo ceee" so I would write "an ROC", if you pronounce it as "rock" you can leave it as is.

https://www.quora.com/When-using-an-abbreviation-starting-with-the-letter-R-is-it-a-or-an-For-example-I-played-an-RTG-game#:~:text=It%20depends%20whether%20you%20pronounce,correct%20to%20use%20a%20instead.